### PR TITLE
[1063]  Document storage

### PR DESCRIPTION
**Pull request by Sybrand**
- Refactoring lots of code out into "wells" from "gwells" app (working in progress).
- Vue component to display additional documents (private and public).
- Fix to keycloak login to stop page reloads.
- Document api (minio) and tests added.
- Additional permissions for viewing private documents.

### CODE REVIEW

**Comment by Sybrand** [comment]
[EPIC#WellSummary | Add links to internal-only documents |5| !!! [business value ranking:25 ]](https://trello.com/c/IzDBNuCY/768-epicwellsummary-add-links-to-internal-only-documents-5-business-value-ranking25)

File: app/frontend/src/wells/components/Documents.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        // The major downside of this is that there's going to be a "popup-blocked" message somewhere.
+        const a = document.createElement('a')
+        a.target = '_blank'
+        a.download = file.name
+        a.href = response.data.url
+        document.body.appendChild(a)
+        a.click()
+      })
+    }
+  }
```
Comment (stephenhillier) [inline app/frontend/src/wells/components/Documents.vue]: Code no longer used

File: app/frontend/src/common/authenticate.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          token,
+          refreshToken,
+          idToken }
+        ).success((result) => {
+          store.commit(SET_KEYCLOAK, instance)
+          if (instance.authenticated) {
+            // We may have been authenticated, but the token could be expired.
+            if (instance.isTokenExpired(0)) {
+              // The token is expired!
+              result = false // As far as we're concerned, this means authentication failed.
```
Comment (stephenhillier) [inline app/frontend/src/common/authenticate.js]: possibly reject promise instead?

File: app/frontend/src/wells/components/Documents.vue (hunk 1 above)
Comment (Sybrand) [reply app/frontend/src/wells/components/Documents.vue]: Removed

File: app/frontend/src/common/authenticate.js (hunk 2 above)
Comment (Sybrand) [reply app/frontend/src/common/authenticate.js]: I'm leaving as is right now - mirroring keycloak having true/false for the init - but it's something to relook. We could probably just get rid of this promise entirely.

**Pull request closed without merging**